pkg/storage/storage.go

```diff
@@ -340,39 +342,59 @@ func (s *storageModule) ensureCache() error {
 
 	var cacheFs filesystem.Volume
 
-	// check if we already have a cache
-	for idx := range s.volumes {
-		filesystems, err := s.volumes[idx].Volumes()
-		if err != nil {
-			return err
-		}
-		for jdx := range filesystems {
-			if filesystems[jdx].Name() == cacheLabel {
-				log.Debug().Msgf("Found existing cache at %v", filesystems[jdx].Path())
-				cacheFs = filesystems[jdx]
+	// check if mounted
+	if !filesystem.IsMountPoint(cacheTarget) {
```

Comment:
I would do this differently actually

```go
if filesystem.IsMoutPoint(cacheTarget) {
return nil
}
```

This I won't have to indent all this code in an if statement

Reply:
fixed